scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2929,7 +2951,10 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   }
 
   def simpleExprRest(t: Term, canApply: Boolean): Term = atPos(t, auto) {
-    if (canApply) newLineOptWhenFollowedBy[LeftBrace]
+    if (canApply) {
+      newLineOptWhenFollowedBy[LeftBrace]
```

Comment:
I played with the original sample and it looks like Scala3 also has a similar rule for `LeftBrace`.

```scala
def fun(a: Int, b: Int, c: Int)(d: Int): String = ???
def method: String = 
  fun(1, 2, 3) 
    {4} // <- this  Term.Apply(Term.Apply("fun", ...), "d")
def method2: String = 
  fun(1, 2, 3)
  {4} // <- doesn't compile - `{4}` is not part of apply
      // but scalameta parses it the same way as above -  Term.Apply(Term.Apply("fun", ...), "d")
```

It seems that this part should be smth like:
```scala
if (canApply) {
  if (dialect.allowSignificantIndentation) {
     // combine these methods in one?
     newLineOptWhenFollowedByIndentationAnd[LeftParen]
     newLineOptWhenFollowedByIndentationAnd[LeftBrace]
  } else {
    newLineOptWhenFollowedBy[LeftBrace]
  }
}
```

Reply:
Good catch! I was mislead a bit by Martin's comment, since he explicitely mentioned `)(`

Thanks! This should be fixed now.